homeassistant/components/husqvarna_automower_ble/__init__.py

```diff
@@ -22,10 +22,14 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Husqvarna Autoconnect Bluetooth from a config entry."""
+    if CONF_PIN not in entry.data:
+        raise ConfigEntryAuthFailed
```

Comment:
This means every single user will be requested to input a PIN, even if it's not required, why do we do that? Can't we detect that a PIN is needed and initiate a reauth flow only then?

Reply:
Also, we don't really have an easy way to detect if it is or isn't required. Note that everyone who sets up a mower knows the PIN, so it isn't really secret

Reply:
We don't have a way to detect a wrong PIN unfortunately. Everything is reverse engineered and detecting a wrong PIN isn't supported at the moment.

Reply:
Ok! We can. I have updated this to detect an invalid PIN from the mower

Reply:
Not that I know of